### PR TITLE
Auction module updated to support currency/hook

### PR DESCRIPTION
**Pull request by jaiminpanchal27**
## Type of change
- [x] Refactoring (no functional changes, no api changes)

## Description of change
Have updated auction module to support currency/hook plugin. 
Moved `isValid` function to validate bid response to bidderFactory.

### CODE REVIEW

File: src/auction.js [hunk 1]
```diff
@@ -353,19 +188,128 @@ export function newAuction({adUnits, adUnitCodes, callback, cbTimeout}) {
...
     getBidsReceived: () => _bidsReceived,
-    callBids
   }
 }
 
+export const addBidResponse = createHook('asyncSeries', function addBidResponse(adUnitCode, bid) {
+  let auctionInstance = this;
+  let bidRequests = auctionInstance.getBidRequests();
+  let auctionId = auctionInstance.getAuctionId();
+
```
Comment (snapwich) [inline src/auction.js]: What happened to `isValid` and all its code that validated the bid?

File: src/auction.js [hunk 2]
```diff
@@ -353,19 +188,128 @@ export function newAuction({adUnits, adUnitCodes, callback, cbTimeout}) {
...
+    } else {
+      doCallbacksIfNeeded();
+      addBidToAuction(bidResponse);
+    }
+  }
+});
+
+// Postprocess the bids so that all the universal properties exist, no matter which bidder they came from.
+// This should be called before addBidToAuction().
+function getPreparedBidForAuction({adUnitCode, bid, bidRequests, auctionId}) {
```
Comment (snapwich) [inline src/auction.js]: How come this was moved outside `addBidResponse`?

File: src/auction.js (hunk 1 above)
Comment (jaiminpanchal27) [reply src/auction.js]: I moved `isValid` to `bidderFactory`. I think that is the best place to validate bid response.

File: src/auction.js (hunk 2 above)
Comment (jaiminpanchal27) [reply src/auction.js]: addBidResponse function had multiple inner functions like `isValid`, `preparedBidForAuction` and `validBidSize` which was creating confusing so moved validation logic to `bidderFactory` and also moved `getPreparedBidForAuction`. 

Previously `preparedBidForAuction` was mutating bid object and now its not.

**Pull request merged**